tests/core/engine_adapter/test_clickhouse.py

```diff
@@ -333,3 +346,443 @@ def test_create_table_properties(make_mocked_engine_adapter: t.Callable, mocker)
+    )
+
+    # User already set the setting, so we should not override it
+    query_with_setting = query.copy()
+    query_with_setting.set(
+        "settings",
+        [
+            exp.EQ(
+                this=exp.var("join_use_nulls"),
+                expression=exp.Literal(this="0", is_string=False),
+            )
+        ],
+    )
+
+    assert (
+        adapter.inject_query_setting(query_with_setting, check_server_default=True)
+        == query_with_setting
+    )
+
+    # Server default of 0 != method default of 1, so we inject 0
+    mocker.patch(
+        "sqlmesh.core.engine_adapter.ClickhouseEngineAdapter.fetchone",
+        return_value="0",
+    )
+
+    assert (
+        adapter.inject_query_setting(query, check_server_default=True).sql(adapter.dialect)
+        == "SELECT col1 FROM table SETTINGS join_use_nulls = 0"
+    )
+
+
+def test_scd_type_2_by_time(
+    make_mocked_engine_adapter: t.Callable, mocker: MockerFixture, make_temp_table_name: t.Callable
+):
+    adapter = make_mocked_engine_adapter(ClickhouseEngineAdapter)
+
+    temp_table_mock = mocker.patch("sqlmesh.core.engine_adapter.EngineAdapter._get_temp_table")
+    table_name = "target"
+    temp_table_id = "abcdefgh"
+    temp_table_mock.return_value = make_temp_table_name(table_name, temp_table_id)
+
+    # The SCD query we build must specify the setting join_use_nulls = 1. We need to ensure that our
+    # setting on the outer query doesn't override the value the user expects.
+    #
+    # This test's user query does not contain a setting "join_use_nulls", so we determine whether or not
+    # to inject it based on the current server value. The mocked server value is 1, so we should not
+    # inject.
+    fetchone_mock = mocker.patch("sqlmesh.core.engine_adapter.ClickhouseEngineAdapter.fetchone")
+    fetchone_mock.return_value = "1"
+
+    adapter.scd_type_2_by_time(
+        target_table="target",
+        source_table=t.cast(
+            exp.Select, parse_one("SELECT id, name, price, test_UPDATED_at FROM source")
+        ),
+        unique_key=[
+            parse_one("""COALESCE("id", '') || '|' || COALESCE("name", '')"""),
+            parse_one("""COALESCE("name", '')"""),
+        ],
+        valid_from_col=exp.column("test_valid_from", quoted=True),
+        valid_to_col=exp.column("test_valid_to", quoted=True),
+        updated_at_col=exp.column("test_UPDATED_at", quoted=True),
+        columns_to_types={
+            "id": exp.DataType.build("INT"),
+            "name": exp.DataType.build("VARCHAR"),
+            "price": exp.DataType.build("DOUBLE"),
+            "test_UPDATED_at": exp.DataType.build("TIMESTAMP"),
+            "test_valid_from": exp.DataType.build("TIMESTAMP"),
+            "test_valid_to": exp.DataType.build("TIMESTAMP"),
+        },
+        execution_time=datetime(2020, 1, 1, 0, 0, 0),
+    )
+
+    assert adapter.cursor.execute.call_args_list[4][0][0] == parse_one(
```

Comment:
I'm curious - did `to_sql_calls(adapter)` not work here?

Reply:
It does - oversight on my part